### PR TITLE
feat: add image naming and versioning

### PR DESCRIPTION
**Pull request by liadgo**
Signed-off-by: Liad <liadg@cisco.com>

Fixes #

### CODE REVIEW

**Comment by liadgo** [comment]
@Chriscbr 🙏

File: src/image.ts [hunk 1]
```diff
@@ -49,6 +49,16 @@ export interface ImageProps {
...
 
+  /**
+   * Name for tagging the image
+   */
+  readonly name?: string;
+
+  /**
+   * Tag for tagging the image
+   */
+  readonly tag?: string;
```
Comment (Chriscbr) [inline src/image.ts:60]: The description of these fields might be confusing to someone less familiar with how docker tagging works. From what I see at https://docs.docker.com/engine/reference/commandline/tag/ it looks like the convention is `{registry}/{name}:{tag}` -- maybe we could mention that in the descriptions, or link to those docs explicitly?

Let's also update the docstrings to mention what the defaults are:

```
@default - auto-generated name
...
@default "latest"
```

**Review by Chriscbr — CHANGES_REQUESTED** [review]

File: src/image.ts (hunk 1 above)
Comment (liadgo) [reply src/image.ts]: Got you
just updated!

**Review by Chriscbr — APPROVED** [review]
Thanks @liadgo !

**Pull request merged**